### PR TITLE
[Teacher] Modules List

### PR DESCRIPTION
**Pull request by armstrongnate**
There's a lot of boilerplate here because it's the first time we've actually used `Core` in Teacher. It sets up the AppEnvironment, adds a router, etc.

Adds a `TeacherTests` target for unit tests.

The`GetCustomColors` use case was created anew. I didn't update every presenter to use the new one. I figure we can do that when we update them all as part of ripping out the old Operation-based use cases.

### CODE REVIEW

**Comment by inst-danger** [comment]
Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Teacher** | 92.31% | 92.3% | 0.01%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 91.01% | 91.07% | -0.06%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Student** | 93.08% | 93.12% | -0.04%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Store/Store.swift [hunk 1]
```diff
@@ -84,6 +84,9 @@ public class Store<U: UseCase>: NSObject, NSFetchedResultsControllerDelegate {
     }
 
     public func refresh(force: Bool = false) {
+        if force {
+            pending = true
+        }
         notify() // send cache
         pending = true
```
[the remark this answers is not in the document]
Comment (armstrongnate) [reply Core/Core/Store/Store.swift]: Actually yeah! We can just set `pending` and that will trigger `notify()`. That will clean up nicely.

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please run `yarn update-headers`. The following do not have the correct license header: 
* Core/CoreTests/API/APIModuleRequestableTests.swift
* Core/CoreTests/Database/Models/ColorTests.swift
* Core/CoreTests/Database/Models/ModuleTests.swift
* Core/CoreTests/Use Cases/GetModulesTests.swift
* TestsFoundation/TestsFoundation/Fixtures/Api/APIModuleFixture.swift
* TestsFoundation/TestsFoundation/Fixtures/Model/ModuleFixture.swift
* rn/Teacher/ios/TeacherTests/Modules/ModuleListPresenterTests.swift
* rn/Teacher/ios/TeacherTests/TeacherTestCase.swift
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**